### ISSUE TITLE
Help, When I use SageAttention==1.0.6, got Accelerate up 9% and Accuracy down 7%

### ISSUE THREAD

**Issue by shushanxingzhe** [thu-ml/SageAttention#90]
Hello, while I installed SageAttention==1.0.6 to accelerate my Qwen-audio model training, I got accelerated up 9% against FlashAttention==2.6.3 on my 2 node * 4 L40s cluster. It's really great and thank for your amazing work. But unfortunately, the accuracy dropped from 90.6 % to 83.76%. This is still far from the effect of your propaganda without lossing end-to-end metrics across various models. Am I using it the wrong way? Or you can give me some suggestions to improve the situation.
Thanks in advance @jt-zhang @jason-huang03 @DefTruth

**Comment by jason-huang03** [comment]
Hi, SageAttention is not for training but for inference. How to you manage to train your model with SageAttention?

**Comment by shushanxingzhe** [comment]
> Hi, SageAttention is not for training but for inference. How to you manage to train your model with SageAttention?

I just enable the following code at the start of my training script. So do you have any plan to support training in addition to inference in the future.
`from sageattention import sageattn
import torch.nn.functional as F

F.scaled_dot_product_attention = sageattn`

**Comment by jason-huang03** [comment]
Hi, sage attention does not support training so it might be the reason why the accuracy drops (the gradient is cut off).  Supporting training presents a bunch of new difficulties, and we are currently trying to figure out how to do that, so sorry there is no timeline for this feature.

**Comment by jt-zhang** [comment]
Also, we recommend using SageAttention2.0 on L40.
